### PR TITLE
[Excel](custom functions) update tutorial to latest correct steps

### LINKED ISSUE #2936: ContosoAddin.Url
Can you please expleain, why you introduce the "ContosoAddin.Url" element?
Whats it means, what it for?

https://github.com/OfficeDev/office-js-docs-pr/blob/master/docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md

### PR DESCRIPTION
**Pull request by davidchesnut**
The shared runtime tutorial is a bit out of date. I updated it to match the steps in [Configure your Office Add-in to use a shared JavaScript runtime](https://docs.microsoft.com/office/dev/add-ins/develop/configure-your-add-in-to-use-a-shared-runtime). This also fixes #2936

### CODE REVIEW

File: docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md [hunk 1]
```diff
@@ -1,100 +1,132 @@
...
 
-For most custom functions scenarios, we recommend using a shared runtime, unless you have a specific reason to use a non-task pane (UI-less) custom function.
+## Create the add-in project
 
-This tutorial assumes you're familiar with using the Yo Office generator to create add-in projects. Consider completing the [Excel custom functions tutorial](excel-tutorial-create-custom-functions.md), if you haven't already.
+You'll use the [Yeoman generator for Office Add-ins](https://github.com/OfficeDev/generator-office) to create the Excel add-in project.
 
-## Create the add-in project
+- To generate an Excel add-in with custom functions, run the command `yo office --projectType excel-functions --name 'Excel shared runtime add-in' --host excel --js true`.
```
Comment (AlexJerabek) [inline docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md:17]: Can we put this yo office command in the command line code box so it's easily copied?

File: docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md [hunk 2]
```diff
@@ -1,100 +1,132 @@
...
 
-Use the Yeoman generator to create an Excel add-in project. Run the following command and then answer the prompts with the following answers.
+The generator will create the project and install supporting Node components.
```
Comment (AlexJerabek) [inline docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md:19]: Small tense suggestion.
```suggestion
The generator creates the project and installs supporting Node components.
```

File: docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md [hunk 3]
```diff
@@ -1,100 +1,132 @@
...
 
-- Choose a project type: **Excel Custom Functions Add-in project**
-- Choose a script type: **JavaScript**
-- What do you want to name your add-in? **My Office Add-in**
+Follow these steps to configure the add-in project to use a shared runtime.
 
-![Screenshot showing the prompts and answers for the Yeoman generator in a command line interface.](../images/yo-office-excel-project.png)
+1. Start Visual Studio Code and open the add-in project you generated.
+1. Open the **manifest.xml** file.
+1. Update the requirements section to use the [shared runtime](../reference/requirement-sets/shared-runtime-requirement-sets.md) instead of the custom function runtime. The XML should appear as follows.
```
Comment (AlexJerabek) [inline docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md:27]: Should we be explicit that "requirements" is an XML element?
```suggestion
1. Update the `<Requirements>` section to use the [shared runtime](../reference/requirement-sets/shared-runtime-requirement-sets.md) instead of the custom function runtime. The XML should appear as follows.
```

File: docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md [hunk 4]
```diff
@@ -1,100 +1,132 @@
...
+    <Hosts>
+      <Host Name="Workbook"/>
+    </Hosts>
+    <Requirements>
+      <Sets DefaultMinVersion="1.1">
+        <Set Name="SharedRuntime" MinVersion="1.1"/>
+      </Sets>
+    </Requirements>
+    <DefaultSettings>
+    ```
```
Comment (AlexJerabek) [inline docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md:39]: I like having the context for where Requirements is in the file, but I don't like having to copy everything when only the one part is getting replaced.

File: docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md [hunk 5]
```diff
@@ -1,100 +1,132 @@
...
      <Hosts>
-       <Host xsi:type="Workbook">
+       <Host ...>
          <Runtimes>
-           <Runtime resid="ContosoAddin.Url" lifetime="long" />
+           <Runtime resid="Taskpane.Url" lifetime="long" />
          </Runtimes>
-       <AllFormFactors>
+       ...
+       </Host>
```
Comment (AlexJerabek) [inline docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md:54]: It's a little confusing that this `<Host>` element doesn't match the one in the previous step.

File: docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md [hunk 6]
```diff
@@ -1,100 +1,132 @@
...
+The **webpack.config.js** will build multiple runtime loaders. You need to modify it to load only the shared JavaScript runtime via the **taskpane.html** file.
 
-   ```xml
-   <bt:Urls>
-   <bt:Url id="Functions.Script.Url" DefaultValue="https://localhost:3000/dist/functions.js"/>
-   ...
-   <bt:Url id="ContosoAddin.Url" DefaultValue="https://localhost:3000/taskpane.html"/>
-   ...
-   ```
+1. Start Visual Studio Code and open the Excel or PowerPoint add-in project you generated.
```
Comment (AlexJerabek) [inline docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md:83]: You can remove this step since we've already opened VS Code.

File: docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md [hunk 7]
```diff
@@ -1,100 +1,132 @@
...
-8. Save your changes and rebuild the project.
+1. Add **functions** and **commands** entries to the chunks list as shown next.
+
+    ```javascript
+      new HtmlWebpackPlugin({
+        filename: "taskpane.html",
+        template: "./src/taskpane/taskpane.html",
+        chunks: ["polyfill", "taskpane", "commands", "functions"]
+      })
+    ```
```
Comment (AlexJerabek) [inline docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md:113]: Consider describing the changes, then providing the final `plugins` code bock. That would make it quicker to copy and paste the correct code.

File: docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md [hunk 8]
```diff
@@ -1,100 +1,132 @@
...
+
+1. Save your changes and rebuild the project.
 
    ```command line
    npm run build
    ```
 
+> [!NOTE]
+> You can also remove the **functions.html** and **commands.html** files. The **taskpane.html** will load the **functions.js** and **commands.js** code into the shared JavaScript runtime via the webpack updates you just made.
```
Comment (AlexJerabek) [inline docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md:122]: This note isn't indented under the previous step, which is also causing the next step to appear as "1".

image

File: docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md [hunk 9]
```diff
@@ -1,100 +1,132 @@
...
+
+1. Save your changes and run the project. Ensure that it loads and runs with no errors.
+
+   ```command line
+   npm start
```
Comment (AlexJerabek) [inline docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md:127]: I got this error in the workbook that opens for this step:
image

**Review by AlexJerabek — CHANGES_REQUESTED** [review]
I hit an error halfway through. Any ideas? Maybe someone else could try?

File: docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md (hunk 1 above)
Comment (davidchesnut) [reply docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md:17]: good idea!

File: docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md (hunk 4 above)
Comment (davidchesnut) [reply docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md:39]: It's tricky because over time we've had issues where people get confused about where to put some of the XML, or accidentally put it inside another tag and such. So I've added some context about where it is located to try to help. But yeah, then you can't just copy paste...

File: docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md (hunk 5 above)
Comment (davidchesnut) [reply docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md:54]: Well, unfortunately, that is how the manifest is designed. This is actually a different section. Again why I try to include more details to help know where you are making updates.

Comment (AlexJerabek) [reply docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md:54]: ... I may have messed up this manifest, which would explain the error. Let me try again, but better!

File: docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md (hunk 9 above)
Comment (AlexJerabek) [reply docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md:127]: Okay, fixing the manifest fixed this error. Shocking, I know...

File: docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md [hunk 10]
```diff
@@ -212,3 +244,7 @@ Once Excel starts, you can use the task pane buttons to store or get shared data
```
Comment (AlexJerabek) [inline docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md:244]: Small nit, but in the last step we're asked to call `npm run start`. Two things:
- We called `npm start` earlier. Should this be the same command?
- The tutorial never has the user close the add-in. Should we have them close and reopen? Or should we remove the earlier start? FWIW, I never closed the previous run and didn't get the new functions. Closing and rerunning npm start refreshed things.

File: docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md [hunk 11]
```diff
@@ -1,99 +1,154 @@
...
+
+1. Save the **manifest.xml** file.
+
+## Configure the webpack.config.js file
+
+The **webpack.config.js** will build multiple runtime loaders. You need to modify it to load only the shared JavaScript runtime via the **taskpane.html** file.
+
+1. Open the **webpack.config.js** file.
+1. Go to the `plugins:` section.
+1. Remove the following `functions.html` plugin if it esists
```
Comment (AlexJerabek) [inline docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md:108]:
```suggestion
1. Remove the following `functions.html` plugin if it exists.
```

File: docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md [hunk 12]
```diff
@@ -1,99 +1,154 @@
...
+      new HtmlWebpackPlugin({
+        filename: "taskpane.html",
+        template: "./src/taskpane/taskpane.html",
+        chunks: ["polyfill", "taskpane", "commands", "functions"]
+      })
+    ```
+    
+1. Save your changes and rebuild the project.
 
    ```command line
```
Comment (AlexJerabek) [inline docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md:140]:
```suggestion
   ```command&nbsp;line
```

File: docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md [hunk 13]
```diff
@@ -1,99 +1,154 @@
...
    npm run build
    ```
+    
+    > [!NOTE]
+    > You can also remove the **functions.html** and **commands.html** files. The **taskpane.html** will load the **functions.js** and **commands.js** code into the shared JavaScript runtime via the webpack updates you just made.
+    
+1. Save your changes and run the project. Ensure that it loads and runs with no errors.
+    
+   ```command line
```
Comment (AlexJerabek) [inline docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md:149]:
```suggestion
   ```command&nbsp;line
```

File: docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md [hunk 14]
```diff
@@ -1,99 +1,154 @@
...
+    ```
+    
+1. Save your changes and rebuild the project.
 
    ```command line
    npm run build
    ```
+    
+    > [!NOTE]
+    > You can also remove the **functions.html** and **commands.html** files. The **taskpane.html** will load the **functions.js** and **commands.js** code into the shared JavaScript runtime via the webpack updates you just made.
```
Comment (AlexJerabek) [inline docs/tutorials/share-data-and-events-between-custom-functions-and-the-task-pane-tutorial.md:145]:
```suggestion
    > You can also remove the **functions.html** and **commands.html** files. The **taskpane.html** loads the **functions.js** and **commands.js** code into the shared JavaScript runtime via the webpack updates you just made.
```
